src/features/impl_rand.rs

```diff
@@ -84,10 +84,11 @@ macro_rules! impl_float_types {
         impl Distribution<$quat> for Standard {
             #[inline]
             fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> $quat {
-                let yaw = -PI + rng.gen::<$t>() * 2.0 * PI;
-                let pitch = -PI + rng.gen::<$t>() * 2.0 * PI;
-                let roll = -PI + rng.gen::<$t>() * 2.0 * PI;
-                $quat::from_euler(crate::EulerRot::YXZ, yaw, pitch, roll)
+                let u = rng.gen_range::<$t, _>(0.0..=1.0);
+                let (u1, u2) = ($t::sqrt(1.0 - u), $t::sqrt(u));
+                let (sin1, cos1) = rng.gen_range::<$t, _>(0.0..=TAU).sin_cos();
+                let (sin2, cos2) = rng.gen_range::<$t, _>(0.0..=TAU).sin_cos();
```

Comment:
Regarding the test fail, these need to be changed to `math::sin_cos(rng.gen_range::<$t, _>(0.0..=TAU))`

`$t::sqrt` also needs to change to `math::sqrt`, you'll also need to add a `use crate::f32::math` or `use crate::f64::math` in the relevant `mod` block.